sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -251,26 +258,29 @@ statement
     | (MSCK)? REPAIR TABLE identifierReference
         (option=(ADD|DROP|SYNC) PARTITIONS)?                           #repairTable
     | op=(ADD | LIST) identifier .*?                                   #manageResource
-    | SET COLLATION collationName=identifier                           #setCollation
-    | SET ROLE .*?                                                     #failNativeCommand
+    | CREATE INDEX (IF errorCapturingNot EXISTS)? identifier ON TABLE?
+        identifierReference (USING indexType=identifier)?
+        LEFT_PAREN columns=multipartIdentifierPropertyList RIGHT_PAREN
+        (OPTIONS options=propertyList)?                                #createIndex
+    | DROP INDEX (IF EXISTS)? identifier ON TABLE? identifierReference #dropIndex
+    | unsupportedHiveNativeCommands .*?                                #failNativeCommand
+    ;
+
+setResetStatement
+    : SET COLLATION collationName=identifier                           #setCollation
+    | SET ROLE .*?                                                     #failSetRole
     | SET TIME ZONE interval                                           #setTimeZone
     | SET TIME ZONE timezone                                           #setTimeZone
     | SET TIME ZONE .*?                                                #setTimeZone
     | SET (VARIABLE | VAR) assignmentList                              #setVariable
     | SET (VARIABLE | VAR) LEFT_PAREN multipartIdentifierList RIGHT_PAREN EQ
-          LEFT_PAREN query RIGHT_PAREN                                 #setVariable
+        LEFT_PAREN query RIGHT_PAREN                                   #setVariable
```

Comment:
can we avoid duplicating the SET VAR parser rule?

Reply:
Reasons for "duplication" (quotes because it's not exactly the same):
- In standalone SQL statements, VAR or VARIABLE is mandatory
- In SQL scripts, VAR or VARIABLE is not mandatory
- In SQL scripts, we still want to allow VAR or VARIABLE to be "compatible" with standalone SQL statements

This means that we can either:
- Have one parser rule for set variable statements and reuse it, but handle this logic (and potential syntax exceptions) in the visitor functions
- Have this handled in the grammar (one rule has ? where VAR or VARIABLE is not mandatory, the other doesn't)

We decided on the second approach, but can change to the first one if you think so.

Reply:
I prefer the first option as it's more explicit. For example I missed this subtle difference in the ANTLR rules.

The code structure can be
```
setVariable
    : SET (VARIABLE | VAR)? assignmentList
    | ...

compoundStatement
    : statement
    | setVariable
    | ...

setResetStatement
    : ...
    | setVariable
    | ...
```